### PR TITLE
Update hydra chart to 0.23.2

### PR DESCRIPTION
**Pull request by werdes72**
https://github.com/kyma-project/kyma/issues/14266

### CODE REVIEW

File: resources/ory/charts/hydra/charts/hydra-maester/values.yaml [hunk 1]
```diff
@@ -1,24 +1,28 @@
+# -- Number of replicas in deployment
```
Comment (cnvergence) [inline resources/ory/charts/hydra/charts/hydra-maester/values.yaml:1]: Hydra-maester update changes will be added in separate PR: https://github.com/kyma-project/kyma/pull/14301

File: resources/ory/charts/hydra/charts/hydra-maester/crds/crd-oauth2clients.yaml [hunk 2]
```diff
@@ -0,0 +1,199 @@
+
```
Comment (cnvergence) [inline resources/ory/charts/hydra/charts/hydra-maester/crds/crd-oauth2clients.yaml:1]: We keep CRDs in kyma/installation/resources/crds/ory, so I think this is not needed

**Review by cnvergence — CHANGES_REQUESTED** [review]

File: resources/ory/charts/oathkeeper/Chart.yaml [hunk 3]
```diff
@@ -24,4 +24,5 @@ type: application
 dependencies:
   - name: oathkeeper-maester
     version: 0.23.1
-    condition: oathkeeper-maester.enabled
+    condition: maester.enabled
```
Comment (cnvergence) [inline resources/ory/charts/oathkeeper/Chart.yaml:27]: This change would also come as a separate PR

File: resources/ory/charts/hydra/values.yaml [hunk 4]
```diff
@@ -206,19 +271,126 @@ deployment:
...
+
+  # -- If you want to add lifecycle hooks.
+  lifecycle: {}
+  # lifecycle: |
+  #   preStop:
+  #     exec:
+  #       command: [...]
+
+  # -- Set automounting of the SA token
+  automountServiceAccountToken: true
```
Comment (barchw) [inline resources/ory/charts/hydra/values.yaml:327]: Same as: https://github.com/kyma-project/kyma/pull/14301#issuecomment-1128530248

File: resources/ory/charts/hydra/values.yaml [hunk 5]
```diff
@@ -206,19 +271,126 @@ deployment:
...
+  # -- Set sharing process namespace
+  shareProcessNamespace: false
+
+  # -- Specify the serviceAccountName value.
+  # In some situations it is needed to provides specific permissions to Hydra deployments
+  # Like for example installing Hydra on a cluster with a PosSecurityPolicy and Istio.
+  # Uncoment if it is needed to provide a ServiceAccount for the Hydra deployment.
+  serviceAccount:
+    # -- Specifies whether a service account should be created
+    create: true
```
Comment (barchw) [inline resources/ory/charts/hydra/values.yaml:338]: Should this be created by default?

File: resources/ory/charts/hydra/values.yaml [hunk 6]
```diff
@@ -206,19 +271,126 @@ deployment:
...
+pdb:
+  enabled: false
+  spec:
+    minAvailable: 1
+
+#  -- Parameters for the Prometheus ServiceMonitor objects.
+# Reference: https://docs.openshift.com/container-platform/4.6/rest_api/monitoring_apis/servicemonitor-monitoring-coreos-com-v1.html
+serviceMonitor:
+  # -- switch to false to prevent creating the ServiceMonitor
+  enabled: true
```
Comment (barchw) [inline resources/ory/charts/hydra/values.yaml:382]: Do we need this service? Isn't this already done in monitoring Kyma component?

File: resources/ory/charts/oathkeeper/Chart.yaml (hunk 3 above)
Comment (barchw) [inline resources/ory/charts/oathkeeper/Chart.yaml:27]: This should be done in Oathkeeper PR

File: resources/ory/charts/hydra/values.yaml [hunk 7]
```diff
@@ -35,24 +43,50 @@ service:
     type: ClusterIP
     # The service port
     port: 4445
+    # -- The service port name. Useful to set a custom service port name if it must follow a scheme (e.g. Istio)
+    name: http
```
Comment (barchw) [inline resources/ory/charts/hydra/values.yaml:47]: This should probably be commented out

File: resources/ory/charts/hydra/templates/service-admin.yaml [hunk 8]
```diff
@@ -16,12 +21,42 @@ spec:
...
-      name: http
-    - port: 15020
-      targetPort: 15020
-      protocol: TCP
-      name: tcp-status-port
+      name: {{ .Values.service.admin.name }}
   selector:
     app.kubernetes.io/name: {{ include "hydra.name" . }}
     app.kubernetes.io/instance: {{ .Release.Name }}
+{{- if and (.Capabilities.APIVersions.Has "monitoring.coreos.com/v1") (.Values.serviceMonitor.enabled) }}
```
Comment (barchw) [inline resources/ory/charts/hydra/templates/service-admin.yaml:28]: Do we need this service? Isn't this already done in Kyma/monitoring component?

File: resources/ory/charts/hydra/templates/deployment.yaml [hunk 9]
```diff
@@ -40,75 +39,40 @@ spec:
       imagePullSecrets:
         {{- toYaml . | nindent 8 }}
     {{- end }}
-    {{- if .Values.hydra.autoMigrate }}
-      initContainers:
-        {{- if not .Values.global.ory.hydra.persistence.gcloud.enabled }} #Use only for in-cluster databases that may not be ready yet. External databases are made ready before installation.
-        - name: wait-for-db
```
Comment (barchw) [inline resources/ory/charts/hydra/templates/deployment.yaml:46]: Is this OK to be deleted? The wait-for-db.sh script wasn't deleted either way

File: resources/ory/charts/hydra/templates/deployment.yaml [hunk 10]
```diff
@@ -22,8 +22,6 @@ spec:
     matchLabels:
       app.kubernetes.io/name: {{ include "hydra.name" . }}
       app.kubernetes.io/instance: {{ .Release.Name }}
-  strategy:
-    {{- toYaml .Values.deployment.strategy | nindent 4 }}
```
Comment (barchw) [inline resources/ory/charts/hydra/templates/deployment.yaml:26]: Why are we deleting this? Shouldn't that stay and become exposed in values.yaml?

File: resources/ory/charts/hydra/templates/deployment.yaml [hunk 11]
```diff
@@ -144,26 +108,20 @@ spec:
...
+            {{- toYaml .Values.deployment.readinessProbe | nindent 12 }}
           env:
             {{- if .Values.deployment.tracing.datadog.enabled }}
             - name: TRACING_PROVIDER
               value: datadog
             - name: DD_ENV
               value: {{ .Values.deployment.tracing.datadog.env | default "none" | quote }}
             - name: DD_VERSION
-              value: {{ .Values.deployment.tracing.datadog.version | default .Values.global.images.hydra.version | quote }}
+              value: {{ .Values.deployment.tracing.datadog.version | default .Values.image.tag | quote }}
```
Comment (barchw) [inline resources/ory/charts/hydra/templates/deployment.yaml:124]: Do we need to change this?

File: resources/ory/charts/hydra/templates/service-admin.yaml (hunk 8 above)
Comment (cnvergence) [reply resources/ory/charts/hydra/templates/service-admin.yaml:28]: I think service monitor is still needed here

File: resources/ory/charts/hydra/values.yaml (hunk 5 above)
Comment (cnvergence) [reply resources/ory/charts/hydra/values.yaml:338]: Agree, I don't think we need to change the default, which was set to false

File: resources/ory/charts/hydra/values.yaml (hunk 6 above)
Comment (cnvergence) [reply resources/ory/charts/hydra/values.yaml:382]: I think service monitor is still needed here

File: resources/ory/charts/hydra/templates/deployment.yaml (hunk 9 above)
Comment (cnvergence) [reply resources/ory/charts/hydra/templates/deployment.yaml:46]: Yes! Please do not remove wait-for-db

File: resources/ory/charts/hydra/templates/deployment.yaml (hunk 10 above)
Comment (cnvergence) [reply resources/ory/charts/hydra/templates/deployment.yaml:26]: This should stay and is configured by us in the values.yaml

File: resources/ory/charts/hydra/templates/deployment.yaml (hunk 11 above)
Comment (cnvergence) [reply resources/ory/charts/hydra/templates/deployment.yaml:124]: I would not change this

File: resources/ory/charts/hydra/templates/rbac.yaml [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+metadata:
+  name: {{ include "hydra.serviceAccountName" . }}
+  labels:
+    {{- include "hydra.labels" . | nindent 4 }}
+  {{- with .Values.deployment.serviceAccount.annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+
+{{- if .Values.watcher.enabled }}
```
Comment (cnvergence) [inline resources/ory/charts/hydra/templates/rbac.yaml:14]: Same as
> I have removed this sidecar container in this PR https://github.com/kyma-project/kyma/pull/13705
> We are not needing it as we implemented it in the ory-reconciler

File: resources/ory/charts/hydra/templates/job-migration.yaml [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+    spec:
+    {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      serviceAccountName: {{ include "hydra.job.serviceAccountName" . }}
+      automountServiceAccountToken: {{ .Values.job.automountServiceAccountToken }}
+      containers:
+      - name: {{ .Chart.Name }}-automigrate
+        image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
```
Comment (cnvergence) [inline resources/ory/charts/hydra/templates/job-migration.yaml:38]: I think this should be:
`image: "{{ include "imageurl" (dict "reg" .Values.global.containerRegistry "img" .Values.global.images.hydra) }}"`

File: resources/ory/charts/hydra/values.yaml (hunk 4 above)
Comment (cnvergence) [reply resources/ory/charts/hydra/values.yaml:327]: More info here: 
https://github.com/kyma-project/kyma/issues/14266#issuecomment-1128797835

File: resources/ory/charts/hydra/charts/hydra-maester/values.yaml (hunk 1 above)
Comment (cnvergence) [reply resources/ory/charts/hydra/charts/hydra-maester/values.yaml:1]: Please remove charts/hydra-maester from this PR

File: resources/ory/charts/hydra/values.yaml [hunk 14]
```diff
@@ -35,24 +43,50 @@ service:
...
     # kubernetes.io/ingress.class: nginx
     # kubernetes.io/tls-acme: "true"
+    labels: {}
+    #      If you do want to specify additional labels, uncomment the following
+    #      lines, adjust them as necessary, and remove the curly braces after 'labels:'.
+    #      e.g.  app: hydra
+
+secret:
+  # -- switch to false to prevent creating the secret
+  enabled: true
```
Comment (cnvergence) [inline resources/ory/charts/hydra/values.yaml:60]: let's set it to false, we are creating those secrets in the reconciler

**Review by cnvergence — CHANGES_REQUESTED** [review]

File: resources/ory/charts/hydra/templates/service-admin.yaml (hunk 8 above)
Comment (cnvergence) [reply resources/ory/charts/hydra/templates/service-admin.yaml:28]: Correction, they are placed in ory/templates, so we can remove servicemonitors
